app/eg022_kba_authentication.py

```diff
@@ -0,0 +1,158 @@
+	            id_check_configuration_name='ID Check',
+                require_id_lookup='true',
+                recipient_id="1", 
+                routing_order="1"
+            )
+            # Create your signature tab
+            sign_here1 = SignHere(
+                        name="SignHereTab",
+                        x_position="75",
+                        y_position="572",
+                        tab_label="SignHereTab",
+                        page_number="1",
+                        document_id="1",
+                        # A 1- to 8-digit integer or 32-character GUID to match recipient IDs on your own systems.
+                        # This value is referenced in the Tabs element below to assign tabs on a per-recipient basis.
+                        recipient_id="1" # represents your {RECIPIENT_ID}
+                    )
+
+            # Add the tabs model (including the sign_here tabs) to the signer
+            # The Tabs object wants arrays of the different field/tab types
+            signer1.tabs = Tabs(sign_here_tabs=[sign_here1])
+
+            # Tabs are set per recipient
+            envelope_definition.recipients = Recipients(signers=[signer1])
+            # Step 4: Call the eSignature REST API
+            envelope_api = EnvelopesApi(api_client)
+            results = envelope_api.create_envelope(args['account_id'], envelope_definition=envelope_definition)
+            envelope_id = results.envelope_id
+            app.logger.info(f'Envelope was created. EnvelopeId {envelope_id} ')
+
+            return render_template('example_done.html',
+                        title="Envelope sent",
+                        h1="Envelope sent",
+                        message=f"""The envelope has been created and sent!<br/>
+                        Envelope ID {envelope_id}."""
+            )
+
+        except ApiException as err:
+            error_body_json = err and hasattr(err, 'body') and err.body
+            # we can pull the DocuSign error code and message from the response body
+            error_body = json.loads(error_body_json)
+            error_code = error_body and 'errorCode' in error_body and error_body['errorCode']
+            error_message = error_body and 'message' in error_body and error_body['message']
+            # In production, may want to provide customized error messages and
+            # remediation advice to the user.
+            return render_template('error.html',
+                                   err=err,
+                                   error_code=error_code,
+                                   error_message=error_message
+                                   )
+ 
+     
+
+    else:
+        flash('Sorry, you need to re-authenticate.')
+        # We could store the parameters of the requested operation
+        # so it could be restarted automatically.
+        # But since it should be rare to have a token issue here,
```

Comment:
"... token issued here"

Reply:
no, saying 'it should be rare to have a token issue here' is accurate... there isn't a token issued (generated, provisioned), there is a potential issue or problem and if so, then we present to the enduser that they need to log back in.